src/machine/machinectl.c

```diff
@@ -2133,6 +2134,66 @@ static int import_raw(int argc, char *argv[], void *userdata) {
         return transfer_image_common(bus, m);
 }
 
+static int import_fs(int argc, char *argv[], void *userdata) {
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *m = NULL;
+        _cleanup_close_ int fd = -1;
+        const char *local = NULL, *path = NULL;
+        sd_bus *bus = userdata;
+        int r;
+
+        assert(bus);
+
+        if (argc >= 2)
+                path = argv[1];
+        if (isempty(path) || streq(path, "-"))
+                path = NULL;
+
+        if (argc >= 3)
+                local = argv[2];
+        else if (path)
+                local = basename(path);
+        if (isempty(local) || streq(local, "-"))
+                local = NULL;
+
+        if (!local) {
+                log_error("Need either path or local name.");
+                return -EINVAL;
+        }
+
+        if (!machine_name_is_valid(local)) {
+                log_error("Local name %s is not a suitable machine name.", local);
+                return -EINVAL;
+        }
+
+        if (path) {
+                fd = open(path, O_DIRECTORY|O_RDONLY|O_CLOEXEC);
+                if (fd < 0)
+                        return log_error_errno(errno, "Failed to open directory '%s': %m", path);
+        }
```

Comment:
If I give any path that exists, including empty, it'll be "imported". But later on, nspawn and machined will of course fail to start such a container. So it'd be nice to very here, before beginning of the transfer, that the path is a valid fs image path. An RFE.

Reply:
I had that in, but then removed it again, because of the tree mangling: it would kinda replicate the tree rearrangement stuff a second time, but in a "dry" fashion. But yeah, we could still do this, conceptually it would make a ton of sense.

Another option might be to simply warn loudly after the import.

Reply:
(will also add to TODO)

Reply:
I think it makes sense to do an "after import" fixup if the we are doing a live import and we cannot say if the import was good before it's finished. When we are simply copying a directory, I think it'd make a lot of sense to fail early. In particular, it's easy to give some short directory name, and this just makes for better UX.